src/server/JasmineGraphServer.cpp

```diff
@@ -2280,6 +2347,269 @@ bool JasmineGraphServer::initiateClient(std::string host, int port, int dataPort
+          server->h_length);
+    serv_addr.sin_port = htons(port);
+    if (connect(sockfd, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
+        std::cerr << "ERROR connecting" << std::endl;
+        //TODO::exit
+    }
+
+    bzero(data, FED_DATA_LENGTH);
+    write(sockfd, JasmineGraphInstanceProtocol::HANDSHAKE.c_str(), JasmineGraphInstanceProtocol::HANDSHAKE.size());
+    server_logger.log("Sent : " + JasmineGraphInstanceProtocol::HANDSHAKE, "info");
+    bzero(data, FED_DATA_LENGTH);
+    read(sockfd, data, FED_DATA_LENGTH);
+    string response = (data);
+
+    response = utils.trim_copy(response, " \f\n\r\t\v");
+
+    if (response.compare(JasmineGraphInstanceProtocol::HANDSHAKE_OK) == 0) {
+        server_logger.log("Received : " + JasmineGraphInstanceProtocol::HANDSHAKE_OK, "info");
+        string server_host = utils.getJasmineGraphProperty("org.jasminegraph.server.host");
+        write(sockfd, server_host.c_str(), server_host.size());
+        server_logger.log("Sent : " + server_host, "info");
+
+        write(sockfd, JasmineGraphInstanceProtocol::INITIATE_AGG.c_str(),
+              JasmineGraphInstanceProtocol::INITIATE_AGG.size());
+        server_logger.log("Sent : " + JasmineGraphInstanceProtocol::INITIATE_AGG, "info");
+        bzero(data, FED_DATA_LENGTH);
+        read(sockfd, data, FED_DATA_LENGTH);
+        response = (data);
+        response = utils.trim_copy(response, " \f\n\r\t\v");
+        std::cout << response << std::endl;
+        if (response.compare(JasmineGraphInstanceProtocol::OK) == 0) {
+            server_logger.log("Received : " + JasmineGraphInstanceProtocol::OK, "info");
+            write(sockfd, (trainingArgs).c_str(), (trainingArgs).size());
+            server_logger.log("Sent : training args " + trainingArgs, "info");
+            bzero(data, FED_DATA_LENGTH);
+            return 0;
+            }
+    } else {
+        server_logger.log("There was an error in the invoking training process and the response is :: " + response,
+                        "error");
+    }
+
+    close(sockfd);
+    return 0;
+}
+
+bool JasmineGraphServer::initiateOrgServer(std::string host, int port, int dataPort,std::string trainingArgs,int iteration, string partCount) {
+
+    Utils utils;
+    bool result = true;
+    std::cout << pthread_self() << " host : " << host << " port : " << port << " DPort : " << dataPort << std::endl;
+    int sockfd;
+    char data[FED_DATA_LENGTH];
+    bool loop = false;
+    socklen_t len;
+    struct sockaddr_in serv_addr;
+    struct hostent *server;
+
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+    if (sockfd < 0) {
+        std::cerr << "Cannot accept connection" << std::endl;
+        return 0;
+    }
+
+    if (host.find('@') != std::string::npos) {
+        host = utils.split(host, '@')[1];
+    }
+
+    server = gethostbyname(host.c_str());
+    if (server == NULL) {
+        std::cerr << "ERROR, no host named " << server << std::endl;
```

Comment:
Use logging framework

Reply:
Fixed in 3838b63e0c10352eee23ceff37cafc3492b3c14e

Reply:
Done